src/cli_util.rs

```diff
@@ -2279,8 +2279,14 @@ fn handle_early_args(
     args: &[String],
     layered_configs: &mut LayeredConfigs,
 ) -> Result<(), CommandError> {
-    // ignore_errors() bypasses errors like "--help" or missing subcommand
-    let early_matches = app.clone().ignore_errors(true).get_matches_from(args);
+    // ignore_errors() bypasses errors like missing subcommand
+    let early_matches = app
+        .clone()
+        .disable_version_flag(true)
+        .disable_help_flag(true)
```

Comment:
I see this is tested by `tests/test_global_opts.rs:test_early_args()`, which is probably how you found about it too :)

Reply:
Exactly, that's what failed in https://github.com/martinvonz/jj/pull/1765